src/main/java/duke/ui/Ui.java

```diff
@@ -0,0 +1,123 @@
+package duke.ui;
+
+import duke.task.Task;
+import duke.task.TaskList;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * The user interaction reactor.
+ */
+public class Ui {
+    private Scanner scanner = new Scanner(System.in); //The scanner scans user command.
+
+    /**
+     * Prints welcome message.
+     */
+    public void showWelcome() {
+        System.out.println("____________________________________________________________");
+        System.out.println("Hello! I'm Duke");
+        System.out.println("What can I do for you?");
+        System.out.println("____________________________________________________________");
+    }
+
+    /**
+     * Prints line.
+     */
+    public void showLine() {
+        System.out.println("____________________________________________________________");
+    }
+
+    /**
+     * Reads user command.
+     *
+     * @return User's command.
+     */
+    public String readCommand() {
+        return scanner.nextLine();
+    }
+
+    /**
+     * Prints loading error message.
+     */
+    public void showLoadingError() {
```

Comment:
It may be good to remove the showLoadingError() method and use a throw error statement in the respective method of the **Storage** class instead.